File: fixtures/java-maven/src/main/java/com/example/MathUtils.java
package com.example;

/**
 * Simple math utility class for testing.
 */
public class MathUtils {

    public static int add(int a, int b) {
        return a + b;
    }

    public static int multiply(int a, int b) {
        return a * b;
    }

    public static int divide(int a, int b) {
        if (b == 0) {
            throw new IllegalArgumentException("Division by zero");
        }
        return a / b;
    }

    public static long fibonacci(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("Negative index not allowed");
        }
        if (n == 0) return 0;
        if (n == 1) return 1;
        return fibonacci(n - 1) + fibonacci(n - 2);
    }

    public static boolean isPalindrome(String s) {
        String cleaned = s.toLowerCase().replaceAll("[^a-z0-9]", "");
        return cleaned.equals(new StringBuilder(cleaned).reverse().toString());
    }
}

File: fixtures/php/src/MathUtils.php
<?php

namespace Example;

class MathUtils
{
    public static function add(int $a, int $b): int
    {
        return $a + $b;
    }

    public static function multiply(int $a, int $b): int
    {
        return $a * $b;
    }

    public static function divide(int $a, int $b): int
    {
        if ($b === 0) {
            throw new \InvalidArgumentException("Division by zero");
        }
        return intdiv($a, $b);
    }

    public static function fibonacci(int $n): int
    {
        if ($n < 0) {
            throw new \InvalidArgumentException("Negative index not allowed");
        }
        if ($n === 0) return 0;
        if ($n === 1) return 1;
        return self::fibonacci($n - 1) + self::fibonacci($n - 2);
    }

    public static function isPalindrome(string $s): bool
    {
        $cleaned = strtolower(preg_replace('/[^a-z0-9]/', '', $s));
        return $cleaned === strrev($cleaned);
    }
}
